src/op_handlers/opcode_decommit.rs

```diff
@@ -0,0 +1,45 @@
+use crate::{
+    address_operands::{address_operands_read, address_operands_store},
+    eravm_error::{EraVmError, HeapError},
+    state::VMState,
+    store::Storage,
+    value::{FatPointer, TaggedValue},
+    Opcode,
+};
+
+pub fn opcode_decommit(
+    vm: &mut VMState,
+    opcode: &Opcode,
+    storage: &mut dyn Storage,
+) -> Result<(), EraVmError> {
+    let (src0, src1) = address_operands_read(vm, opcode)?;
+
+    let (code_hash, extra_cost) = (src0.value, src1.value.low_u32());
+
+    let preimage_len_in_bytes = zkevm_opcode_defs::system_params::NEW_KERNEL_FRAME_MEMORY_STIPEND;
+
+    vm.decrease_gas(extra_cost)?;
+
+    let code = storage
+        .decommit(code_hash)?
+        .ok_or(EraVmError::DecommitFailed)?;
+
+    let id = vm.heaps.allocate();
+    let heap = vm.heaps.get_mut(id).ok_or(HeapError::StoreOutOfBounds)?;
+    let mut address = 0;
+    for value in code.iter() {
+        heap.store(address, *value);
+        address += 32;
+    }
```

Comment:
Im the one who did this but...
Maybe we need to do a `heap.expand_memory` here and consume the gas needed

Reply:
Done